### DISCUSSION TITLE
Latex Export

### DISCUSSION THREAD

**Discussion by mariusroets** [niklasf/python-chess#819] [Ideas]
I am creating a publication from PGN by converting to Latex. I have managed to accomplish the task by extending chess.pgn.StringExporter and reimplementing where needed. My question to the developer(s) is: would you be interested in merging this in python-chess?

My second question is that I was not sure what was the best way to do this. I extended StringExporter, but it is also possible to extend StringExporterMixin and BaseVisitor (like StringExporter does), or to create a LatexExporterMixin, and extend that together with BaseVisitor.

Many of my PGN games have comments before the first move of the mainline. Does the package support that?

**Comment by niklasf** [comment]
Hi, I am hesitant, because there are so many different ways to typeset a game, and likely everyone would want something slightly different.

The hierarchy of the exporter classes doesn't matter much. The point of `StringExporterMixin` is that it doesn't determine the `T` in `BaseVisitor[T]`, but that only matters when using type hints, and when you want something other than `str`.

> Many of my PGN games have comments before the first move of the mainline. Does the package support that?

Yes, for `game = chess.pgn.Game()` that would be `game.comment`.
